tests/test_components/test_autograd.py

```diff
@@ -533,8 +563,103 @@ def postprocess(data: td.SimulationData) -> float:
     return dict(sim=make_sim, postprocess=postprocess)
 
 
-@pytest.mark.parametrize("structure_key, monitor_key", args)
-def test_autograd_objective(use_emulated_run, structure_key, monitor_key):
+@pytest.mark.parametrize("axis", (0, 1, 2))
+def test_polyslab_axis_ops(axis):
+    vertices = ((0, 0), (0, 1), (1, 1), (1, 0))
+    p = td.PolySlab(vertices=vertices, axis=axis, slab_bounds=(0, 1))
+
+    ax_coords = np.array([0, 1, 2, 3])
+    plane_coords = np.array([[4, 5], [6, 7], [8, 9], [10, 11]])
+    coord = p.unpop_axis_vect(ax_coords=ax_coords, plane_coords=plane_coords)
+
+    assert np.all(coord[:, axis] == ax_coords)
+
+    _ax_coords, _plane_coords = p.pop_axis_vect(coord=coord)
+
+    assert np.all(_ax_coords == ax_coords)
+    assert np.all(_plane_coords == plane_coords)
+
+    vertices_next = np.roll(vertices, axis=0, shift=-1)
+    edges = vertices_next - vertices
+
+    basis_vecs = p.edge_basis_vectors(edges=edges)
+
+
+@pytest.mark.parametrize("structure_key, monitor_key", (("medium", "mode"),))
+def _test_autograd_numerical(structure_key, monitor_key):
```

Comment:
Is this test not supposed to be run?

Reply:
Yeah we should probably introduce pytest marks (slow, web, etc…) to mark test sets and then selectively run them, with some sane default sets

Reply:
could run the full test suite then only before release for example, or on a big merge

Reply:
That would be good. Although the numerical testing will still be extremely long if we test every combination of structure and monitor. So I'm not sure we want to do it in an automated way. Instead we could leave it for individual feature testing

Reply:
not parametrization, just custom marks. they are basically labels for tests, and then you can selectively run specific labels (e.g. all the tests that are labelled as „slow“)
i agree with not running every possible combination

Reply:
ah ok, so here's what I did
```
RUN_NUMERICAL = False
...
@pytest.mark.skipif(not RUN_NUMERICAL, reason="Numerical gradient tests runs through web API.")
...
def test_numerical
```

That way it will skip it by default and we can change this behavior by setting `RUN_NUMERICAL`.

I didnt see a way to skip a test by default and trigger it by command line argument. how does this sound?

Reply:
yeah this is fine for now. what i mean by marks is [this](https://docs.pytest.org/en/stable/example/markers.html), i.e.:
```python
@pytest.mark.slow  # or for example @pytest.mark.numerical
def test_XX(): ...
```
where `slow` or `numerical` are custom marks registered with pytest that we define (in`pytest.ini` or `pyproject.toml`, and then you can run `pytest -v -m slow`, which will run only the tests marked `slow`.
i think this is something to tackle in a larger test refactor though, so i'll just add it to the pile